packages/vesting-base/src/contract.rs

```diff
@@ -0,0 +1,448 @@
+    /// depending on the received template.
+    pub fn execute(
+        &self,
+        deps: DepsMut,
+        env: Env,
+        info: MessageInfo,
+        msg: ExecuteMsg,
+    ) -> Result<Response, ContractError> {
+        match msg {
+            ExecuteMsg::Claim { recipient, amount } => {
+                self.claim(deps, env, info, recipient, amount)
+            }
+            ExecuteMsg::Receive(msg) => self.receive_cw20(deps, env, info, msg),
+            ExecuteMsg::RegisterVestingAccounts { vesting_accounts } => {
+                let config = self.config.load(deps.storage)?;
+
+                match &config.vesting_token {
+                    AssetInfo::NativeToken { denom } if info.sender == config.owner => {
+                        let amount = must_pay(&info, denom)?;
+                        self.register_vesting_accounts(
+                            deps,
+                            vesting_accounts,
+                            amount,
+                            env.block.height,
+                        )
+                    }
+                    _ => Err(ContractError::Unauthorized {}),
+                }
+            }
+            ExecuteMsg::ProposeNewOwner { owner, expires_in } => {
+                let config: Config = self.config.load(deps.storage)?;
+
+                propose_new_owner(
+                    deps,
+                    info,
+                    env,
+                    owner,
+                    expires_in,
+                    config.owner,
+                    &self.ownership_proposal,
+                )
+                .map_err(Into::into)
+            }
+            ExecuteMsg::DropOwnershipProposal {} => {
+                let config: Config = self.config.load(deps.storage)?;
+
+                drop_ownership_proposal(deps, info, config.owner, &self.ownership_proposal)
+                    .map_err(Into::into)
+            }
+            ExecuteMsg::ClaimOwnership {} => claim_ownership(
+                deps,
+                info,
+                env,
+                &self.ownership_proposal,
+                |deps, new_owner| {
+                    self.config.update::<_, StdError>(deps.storage, |mut v| {
+                        v.owner = new_owner;
+                        Ok(v)
+                    })?;
+
+                    Ok(())
+                },
+            )
+            .map_err(Into::into),
+        }
+    }
+
+    /// Receives a message of type [`Cw20ReceiveMsg`] and processes it depending on the received template.
+    ///
+    /// * **cw20_msg** CW20 message to process.
+    fn receive_cw20(
+        &self,
+        deps: DepsMut,
+        env: Env,
+        info: MessageInfo,
+        cw20_msg: Cw20ReceiveMsg,
+    ) -> Result<Response, ContractError> {
+        let config = self.config.load(deps.storage)?;
+
+        // Permission check
+        if cw20_msg.sender != config.owner || token_asset_info(info.sender) != config.vesting_token
```

Comment:
seems this condition is a bit off: in out case 
1. cw20_msg.sender - auction contract
2. config.owner - dao?
3. info.sender - some random user
4. config.vesting_token